handlers/product-move-api/src/main/scala/com/gu/productmove/endpoint/available/AvailableProductMovesEndpoint.scala

```diff
@@ -44,44 +53,121 @@ object AvailableProductMovesEndpoint {
       .serverLogic[TIO] { subscriptionName => run(subscriptionName).tapEither(result => ZIO.log("result tapped: " + result)).map(Right.apply) }
   }
 
+  // sub to test on: "A-S00334930"
   private def run(subscriptionName: String): TIO[OutputBody] =
     runWithEnvironment(subscriptionName).provide(
       AwsS3Live.layer,
       AwsCredentialsLive.layer,
       SttpClientLive.layer,
       ZuoraClientLive.layer,
       GetSubscriptionLive.layer,
+      GetCatalogueLive.layer,
+      GetAccountLive.layer,
       ZuoraGetLive.layer,
       GuStageLive.layer,
     )
 
-  private[productmove] def runWithEnvironment(subscriptionName: String): ZIO[GetSubscription, String, OutputBody] =
+  private val freeTrialDays = 14
+
+
+  def getAvailableSwitchRatePlans(zuoraProductCatalogue: ZuoraProductCatalogue, ratePlanNames: List[String]): List[ZuoraProductRatePlan] =
+    val productRatePlans = for {
+      product <- zuoraProductCatalogue.products.filter(ratePlanNames contains _.name)
+      productRatePlan <- product.productRatePlans.toList
+    } yield productRatePlan
+
+    productRatePlans.find(_.name == "Digital Pack Monthly").toList
+
+  val localDateToString = DateTimeFormatter.ISO_LOCAL_DATE
+
+  def eligibilityCheck(accountNotEligible: Boolean, message: String): ZIO[Any, Unit, Unit] = {
+    if (accountNotEligible) {
+      for {
+        _ <- ZIO.log(message)
+        _ <- ZIO.fail(())
+      } yield ()
+    } else {
+      ZIO.succeed(())
+    }
+  }
```

Comment:
Ignoring the possible scala 3 indentation style for now, but even in scala 2 style we could get rid of most of the braces if you like?
```suggestion
  def eligibilityCheck(accountNotEligible: Boolean, message: String): ZIO[Any, Unit, Unit] = 
    if (accountNotEligible) 
      for {
        _ <- ZIO.log(message)
        _ <- ZIO.fail(())
      } yield ()
    else
      ZIO.succeed(())
```

Also, we have a "Not" in the name of the variable.  Would it be clearer to flip the logic like to this?
```scala
  def eligibilityCheck(accountEligible: Boolean, message: String): ZIO[Any, Unit, Unit] = 
    if (accountEligible) 
      ZIO.succeed(())
    else
      for {
        _ <- ZIO.log(message)
        _ <- ZIO.fail(())
      } yield ()
```

Reply:
@johnduffell thanks, all good suggestions, I'll amend it :)